### ISSUE TITLE
RTCDataChannel not properly closed

### ISSUE THREAD

**Issue by Marcel-G** [algesten/str0m#286]
## Problem
When [RTCDataChannel: close() method](https://developer.mozilla.org/en-US/docs/Web/API/RTCDataChannel/close) on a JS client (Chrome for example) the data channel hangs in `closing` state and never transitions to `closed`.
This causes a memory leak and eventually more data channels cannot be created with the following error:

```
Uncaught (in promise) DOMException: Failed to execute 'createDataChannel' on 'RTCPeerConnection': RTCDataChannel creation failed
```

### Reproduction 

Checkout this branch: https://github.com/Marcel-G/str0m/tree/example/close-data-channel
Run `cargo run --example http-post`
Click Connect, and Run
Observe data channels not entering the closed state

## Possible cause

I suspect the same issue as other libraries:
- https://github.com/pion/sctp/pull/238
- https://github.com/webrtc-rs/webrtc/issues/468

Chrome is waiting for a `outgoingResetRequest` before considering the channel closed.

**Comment by algesten** [comment]
@Marcel-G Thanks for reporting this!

With your repro, I have made a potential fix linked above.

The main problem I see with the fix is that it doesn't close _straight away_. At the points this is detected in the code, it's hard to schedule an immediate write, so the response close waits for the normal RTO, which is 3000ms. Hence I had to up your test cases to like 4000ms wait.

Maybe there's some way to make this faster.
